src/garage/misc/process_samples_utils.py

```diff
@@ -0,0 +1,80 @@
+"""Utiliy functions for batch processing samples."""
+from dowel import tabular
+import numpy as np
+
+from garage import log_performance, TrajectoryBatch
+from garage.misc import tensor_utils
+
+
+def preprocess_samples_for_policy_gradient(itr, paths, max_path_length,
+                                           baseline, discount, env_spec,
+                                           episode_reward_mean):
+    """Return processed sample data based on the collected paths.
+
+    Args:
+        itr (int): Iteration number.
+        paths (list[dict]): A list of collected paths.
+        max_path_length (int): Maximum length of a single rollout.
+        baseline(garage.np.baselines.Baseline): Baseline for GAE
+            (Generalized Advantage Estimation).
+        discount (float): Environment reward discount.
+        env_spec (garage.envs.EnvSpec): Environment specification.
+        episode_reward_mean(collections.deque): Deque to record episode
```

Comment:
please move everything involving `episode_reward_mean` into the algorithm (both maintaining the deque and the logging)